packages/network-controller/src/NetworkController.ts

```diff
@@ -396,25 +562,149 @@ type AutoManagedCustomNetworkClientRegistry = Record<
  * The collection of auto-managed network clients that map to Infura networks
  * as well as custom networks that users have added.
  */
-type AutoManagedNetworkClientRegistry = {
+export type AutoManagedNetworkClientRegistry = {
   [NetworkClientType.Infura]: AutoManagedBuiltInNetworkClientRegistry;
   [NetworkClientType.Custom]: AutoManagedCustomNetworkClientRegistry;
 };
 
+/**
+ * Determines whether the given URL is valid by attempting to parse it.
+ *
```

Comment:
Could this be useful for other controllers?

```suggestion
 *
 * TODO: Move to @metamask/utils.
 *
```

Reply:
Maybe. I haven't seen this anywhere else, and this also relies on a package to do the parsing (`uri-js`), which we may not want to introduce as a dependency to `@metamask/utils`.